src/system.rs

```diff
@@ -348,8 +348,8 @@ impl PendingTypes {
     }
 
     /// Adds parent information storing the name and possible TypeId of the parent
-    pub(crate) fn add_parent(&mut self, name: String) {
-        self.parent = Some((name, self.types_by_id.len()))
+    pub(crate) fn add_parent(&mut self, name: String, type_store: &mut TypeStore) {
+        self.parent = Some((name, self.types_by_id.len() + type_store.types_by_id.len()))
```

Comment:
Can you update either the documentation here or the PR description to explain _why_ this fixes it? I'm not sure I quite understand what's going on here.

Reply:
Previously `type_id` for a parent type was calculated incorrectly resulting to incorrect `Violation`. 
While we are resolving a type definition, the correct `type_id` for any type definition would be `PendingTypes`' `types_by_id` length + `TypeStore`'s `types_by_id` length.
`PendinTypes` is a temporary type storage used while resolving a type definition which then shifts all its contents to `TypeStore`.
`TypeStore` is the common place to store all the types.